Announcement for 0.8.9. (#153)
ethereum/solidity-blog: _posts/2021-09-29-solidity-0.8.9-release-announcement.md

---
layout: post
published: true
title: 'Solidity 0.8.9 Release Announcement'
date: '2021-09-29'
author: Solidity Team
category: Releases
---

> [Solidity v0.8.9](https://github.com/ethereum/solidity/releases/tag/v0.8.9) fixes two important bugs:

Reviewer: ```suggestion
[Solidity v0.8.9](https://github.com/ethereum/solidity/releases/tag/v0.8.9) is a pure bugfix release and fixes two important, but low severity, bugs:
```

Q: Are the bugs "important" when they are both categorized as "very low" in severity?

>  - one related to [immutables of signed integer types shorter than 256 bits](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/) introduced in Solidity 0.6.5

Reviewer: ```suggestion
 - a bug related to [immutables of signed integer types shorter than 256 bits](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/), through which sign extension (cleanup) of its value is not always properly performed, introduced in Solidity 0.6.5
```

Would consider adding short info for people to have context that don't want to read the bug posts.

>  - one related to [immutables of signed integer types shorter than 256 bits](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/) introduced in Solidity 0.5.6.

Reviewer 2: ```suggestion
 - one related to [immutables of signed integer types shorter than 256 bits](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/) introduced in Solidity 0.6.5
```

>  - one introduced with [user defined value types](https://blog.soliditylang.org/2021/09/29/user-defined-value-types-bug/) in Solidity v0.8.8 (released two days ago).

Reviewer: ```suggestion
 - a bug introduced with [user defined value types](https://blog.soliditylang.org/2021/09/29/user-defined-value-types-bug/) in Solidity v0.8.8 (released two days ago), which makes the new feature unnecessarily gas wasteful and may create issues for tooling or contract upgrades.
```
Reading "one" and the sentence structure here felt weird to me but it's just a style question, not super important.

A big shout out to Harry Altman (@haltman-at) of Truffle who discovered the second bug
only some hours after the release of 0.8.8!

> 

Reviewer: ```suggestion

For more details about the bugs, please refer to the respective Security Alert posts linked above.
```

---
layout: post
published: true
title: 'Solidity 0.8.9 Release Announcement'
date: '2021-09-29'
author: Solidity Team
category: Releases
---

[Solidity v0.8.9](https://github.com/ethereum/solidity/releases/tag/v0.8.9) is a pure bugfix release and fixes two important, but low severity, bugs.

> 1. The first bug is related to immutables of signed integer types shorter than 256 bits. The bug causes sign extension (cleanup) of those values to not always being properly performed. It got introduced in Solidity 0.6.5. Read more in the respective [security alert](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/).

Reviewer 3: ```suggestion
1. The first bug is related to immutables of signed integer types shorter than 256 bits. The bug causes sign extension (cleanup) of those values to not always being properly performed. It was introduced in Solidity 0.6.5. Read more in the respective [security alert](https://blog.soliditylang.org/2021/09/29/signed-immutables-bug/).
```

> 2. The second bug got introduced with user defined value types in Solidity v0.8.8 (released two days ago). It makes the new feature unnecessarily gas wasteful and may create issues for tooling or contract upgrades. Find more information in the respective [security alert](https://blog.soliditylang.org/2021/09/29/user-defined-value-types-bug/). A big shout out to Harry Altman (@haltman-at) of Truffle who discovered this bug only some hours after the release of 0.8.8!

Reviewer 3: ```suggestion
2. The second bug was introduced with user defined value types in Solidity v0.8.8 (released two days ago). User defined types with underlying type shorter than 32 bytes used incorrect storage layout and wasted storage. Find more information in the respective [security alert](https://blog.soliditylang.org/2021/09/29/user-defined-value-types-bug/). A big shout out to Harry Altman (@haltman-at) of Truffle who discovered this bug only some hours after the release of 0.8.8!
```

---
layout: post
published: true
title: 'Solidity 0.8.9 Release Announcement'
date: '2021-09-29'
author: Solidity Team
category: Releases
---

[Solidity v0.8.9](https://github.com/ethereum/solidity/releases/tag/v0.8.9) fixes two bugs:

>  - one related to [immutables of signed integer types shorter than 256 bits](TODO LINK)

Reviewer 2: apart from the link also add "introduced in version 0.6.5". This makes it similar in structure to the following summary of the user-defined value type bug.

 - one introduced with [user defined value types](https://blog.soliditylang.org/2021/09/27/user-defined-value-types/) in Solidity v0.8.8 (released yesterday).

The bug has no influence on the correctness of Solidity contracts, but contracts compiled with
Solidity 0.8.8 that use the new feature are unnecessarily wasteful and might have problems
with tooling or contract upgrades.

### Storage Layout of User Defined Value Types

The compiler did not correctly compute the storage layout of user defined value types
based on types that are shorter than 32 bytes. It would always use a full storage slot
for these types, even if the underlying type is shorter.

Types shorter than 32 bytes can share a storage slot with other values if there is
enough space.

> 

Reviewer 2: I feel this section (`Storage Layout of ...`) may be replaced by a sentence such as "For further details about the individual bugs, please read the linked blog posts." Another reason this section may be removed is because we don't have a separate section for the immutable bug in the release blog post.

Author: Yes, we will remove it as soon as we have the blog post about the user defined value types bug.
